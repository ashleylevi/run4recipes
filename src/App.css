.header {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.title {
  font-family: 'Courgette', cursive;
  color: #32636E;
  font-weight: normal;
  font-size: 60px;
  margin-bottom: 0px;
}

.slogan {
  font-family: 'Raleway', sans-serif;
  font-weight: bold;
  font-size: 20px;
  color: #D5AD47;
}

.main-logo {
  width: 18px;
  height: 20px;
}