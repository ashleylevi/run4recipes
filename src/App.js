import React, { Component } from 'react';
import RecipesContainer from './components/RecipesContainer/RecipesContainer';
import { getRecipes } from './utils/apiCalls';
import Nav from '../src/components/Nav/Nav';
import { Route, Switch, Redirect, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import './App.css';


class App extends Component {
  constructor() {
    super()
  }


  render() {
    return (
      <div className="App">
        <div className="header">
          <h1 className="title">run4recipes</h1>
          <h2 className="slogan">HIGH CARB MEAL IDEAS <img src='./images/runner.png' alt="runner" className='main-logo'></img> TO FUEL ANY RUN</h2>
        </div>
        <Nav />
        <RecipesContainer/>
        <Switch>
          <Route 
            exact
            path='/allRecipes' 
            render={() => {
                return <Redirect to='/' />
            }}
          />
          <Route
            path='/pasta'
            component={RecipesContainer}
          />
          <Route
            path='/potatoes'
            component={RecipesContainer}
          />
  
          <Route 
            path='/bread'
            component={RecipesContainer} />
        </Switch>
      </div>
    );
  }
}

export default withRouter(connect(null)(App))
