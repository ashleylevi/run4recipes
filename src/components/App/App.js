import React, { Component } from 'react';
import RecipesContainer from '../RecipesContainer/RecipesContainer';
import favoritesContainer from '../RecipesContainer/RecipesContainer';
import Nav from '../Nav/Nav';
import { Route, Switch, Redirect, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import './App.css';

class App extends Component {
  constructor() {
    super()
  }

  render() {
    return (
      <div className="App">
        <div className="header">
          <h1 className="title">run4recipes</h1>
          <h2 className="slogan">HIGH CARB MEAL IDEAS <img src='./images/runner.png' alt="runner" className='main-logo'></img> TO FUEL ANY RUN</h2>
        </div>
        <Nav />
        <Switch>
          <Route 
            exact
            path='/allRecipes' 
            render={() => {
                return <Redirect to='/' />
            }}
          />
          <Route
            path='/pasta'
            component={RecipesContainer}
          />
          <Route
            path='/potato'
            component={RecipesContainer}
          />
          <Route 
            path='/bread'
            component={RecipesContainer} />
            <Route
              path='/favorites'
              component={favoritesContainer} />
          <Route 
            exact path='/'
            component={RecipesContainer} />
          <Route 
            path='/'
            render={() => {
              return <Redirect to='/' />
          }} />
        </Switch>
      </div>
    );
  }
}

export default withRouter(connect(null)(App))
