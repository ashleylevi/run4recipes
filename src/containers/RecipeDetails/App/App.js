import React, { Component } from 'react';
import RecipesContainer from '../RecipesContainer/RecipesContainer';
import favoritesContainer from '../RecipesContainer/RecipesContainer';
import Nav from '../Nav/Nav';
import { Route, Switch, Redirect, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import './App.css';
import RecipeDetails from '../RecipeDetails/RecipeDetails';

class App extends Component {
  constructor() {
    super()
  }

  render() {
    return (
      <div className="App">
        <div className="header">
          <h1 className="title">run4recipes</h1>
          <h2 className="slogan">HIGH CARB MEAL IDEAS <img src='./images/runner.png' alt="runner" className='main-logo'></img> TO FUEL ANY RUN</h2>
        </div>
        <Nav />
        <Switch>
          <Route 
            exact
            path='/allRecipes' 
            render={() => {
                return <Redirect to='/' />
            }}
          />
          <Route
            exact
            path='/pasta'
            component={RecipesContainer}
            />
          <Route
            exact
            path='/pasta/:name'
            render={({ location }) => {
              return <RecipeDetails />
            }}
            />
          <Route
            exact
            path='/potato'
            component={RecipesContainer}
          />
          <Route
            exact
            path='/potato/:name'
            render={({ location }) => {
              return <RecipeDetails />
            }}
            />
          <Route 
            exact
            path='/bread'
            component={RecipesContainer} 
          />
          <Route
            exact
            path='/bread/:name'
            render={({ location }) => {
              return <RecipeDetails />
            }}
            />
          <Route
            path='/favorites'
            component={RecipesContainer} 
          />
          <Route 
            exact path='/'
            component={RecipesContainer} 
          />
          <Route
            exact
            path='/:name'
            render={({ location }) => {
              return <RecipeDetails />
            }}
            />
          <Route 
            path='/'
            component={RecipesContainer} 
          />
        </Switch>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  pastaRecipes: state.pastaRecipes,
  potatoRecipes: state.potatoRecipes,
  breadRecipes: state.breadRecipes
})

export default withRouter(connect(mapStateToProps)(App))
