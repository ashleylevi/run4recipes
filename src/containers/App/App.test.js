import { mapStateToProps } from './App'
import { shallow } from 'enzyme'
import App from './App'
import React from 'react'

describe('App', () => {
  describe('App component', () => {
    const mockRecipes  = [{
          label: "Spaghetti and meatballs",
          image: "https://image.png",
          ingredientLines: [],
          category: 'bread'
    }]
    it('Should match the snapshot', () => {
      const wrapper = shallow(<App pastaRecipes={mockRecipes} potatoRecipes={mockRecipes} breadRecipes={mockRecipes} />)
      expect(wrapper).toMatchSnapshot()
    })
  })
  
  describe('mapStateToProps', () => {
    it('should return an array with recipe objects', () => {
      const mockRecipes  = [{
        label: "Spaghetti and meatballs",
        image: "https://image.png",
        ingredientLines: [],
        category: 'bread'
      }]
    
      const mockState = {
        pastaRecipes: mockRecipes,
        potatoRecipes: mockRecipes,
        breadRecipes: mockRecipes
      }

      const expected = {
        pastaRecipes: mockRecipes,
        potatoRecipes: mockRecipes,
        breadRecipes: mockRecipes
      }

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
})

