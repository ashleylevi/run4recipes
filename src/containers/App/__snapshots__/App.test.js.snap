// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App component Should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Connect(App))
    breadRecipes={
      Array [
        Object {
          "category": "bread",
          "image": "https://image.png",
          "ingredientLines": Array [],
          "label": "Spaghetti and meatballs",
        },
      ]
    }
    pastaRecipes={
      Array [
        Object {
          "category": "bread",
          "image": "https://image.png",
          "ingredientLines": Array [],
          "label": "Spaghetti and meatballs",
        },
      ]
    }
    potatoRecipes={
      Array [
        Object {
          "category": "bread",
          "image": "https://image.png",
          "ingredientLines": Array [],
          "label": "Spaghetti and meatballs",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
